core/src/main/scala/latis/dataset/CompositeDataset.scala

```
@@ -0,0 +1,82 @@
+package latis.dataset
+
+import cats.data.NonEmptyList
+import cats.effect.IO
+import cats.syntax.all._
+import fs2.Stream
+
+import latis.data.Sample
+import latis.data.StreamFunction
+import latis.metadata.Metadata
+import latis.model.DataType
+import latis.ops.BinaryOperation
+import latis.ops.UnaryOperation
+import latis.util.LatisException
+
+/**
+ * Defines a Dataset with data provided via a list of Datasets.
+ */
+class CompositeDataset(
+  _metadata: Metadata,
+  _model: DataType,
+  datasets: NonEmptyList[Dataset],
+  joinOperation: BinaryOperation,
```

We are going to want to be able to push operations down to member Datasets in the near future. Doing this right for any BinaryOperation is going to be tricky. I suspect we will only support simple append joins for the foreseeable future. Should we leave the teaser here and raise unsupported operation exceptions or build in the append use case and keep it simple?